.travis.yml

```diff
@@ -0,0 +1,40 @@
+language: python
+python: 3.6
+cache: pip
+install:
+  - pip install pipenv
+  - pipenv install --dev
+  - wget https://github.com/Sage-Bionetworks/infra-utils/archive/master.zip -O /tmp/infra-utils.zip
+  - unzip -j -n /tmp/infra-utils.zip -x "infra-utils-master/.gitignore" "infra-utils-master/LICENSE" "infra-utils-master/*.md" "infra-utils-master/aws/*"
+  - ./setup_aws_cli.sh || travis_terminate 1
+  - export SAM_CLI_TELEMETRY=0
+stages:
+  - name: validate
+  - name: test
+    if: branch = master AND type = push
+  - name: deploy
+    if: (tag =~ ^v) OR (branch = master AND type = push)
+jobs:
+  fast_finish: true
+  include:
+    - stage: validate
+      script:
+        - pre-commit autoupdate
+        - pre-commit run --all-files
+        - sam build
+        - python -m pytest tests/ -v
+    - stage: test
+      script:
+        - sam build
+        - |
+          sam package --template-file .aws-sam/build/template.yaml \
+            --s3-bucket essentials-awss3lambdaartifactsbucket-x29ftznj6pqw \
+            --output-template-file .aws-sam/build/set-bucket-tags-macro.yaml
+    - stage: deploy
+      script:
+        - sam build
+        - |
+          sam package --template-file .aws-sam/build/template.yaml \
+            --s3-bucket essentials-awss3lambdaartifactsbucket-x29ftznj6pqw \
+            --output-template-file .aws-sam/build/set-bucket-tags-macro.yaml
+        - aws s3 cp .aws-sam/build/template.yaml s3://bootstrap-awss3cloudformationbucket-19qromfd235z9/$REPO_NAME/$TRAVIS_BRANCH || travis_terminate 1
```

Comment:
travis doesn't provide a `$REPO_NAME` variable.  you can either just set this to `cfn-cr-bucket-tagger` or you can export it in your .travis.yaml file, like this.. https://github.com/Sage-Bionetworks/aws-infra/blob/master/deploy-templates.sh#L5

Reply:
done (I used the travis method of setting environment variables instead)

Reply:
you don't need the `|| travis_terminate 1` when executing as a one-liner.  that's only needed when running *.sh scripts

Reply:
removed